pkg/system/system.go

```diff
@@ -316,6 +318,11 @@ func (sc *Controller) upgradeNydusDaemon(d *daemon.Daemon, c upgradeRequest) err
 		return err
 	}
 
+	su := manager.SupervisorSet.GetSupervisor(d.ID())
+	if err := su.SendStatesTimeout(time.Second * 10); err != nil {
```

Comment:
Why do we have to receive the old nydusd's states before a takeover request to new nydusd ?

Reply:
It's a similar procedure to doDaemonFailover function. The snapshotter needs to send the old nydusd's state to new one when takeover. 

Here send refers to from snapshotter to nydus, as opposed to sendfd process, which may be a little confusing.